src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpgradeReadyCheck.cs

```diff
@@ -23,5 +23,13 @@ public interface IUpgradeReadyCheck
         /// <param name="token">A cancellation token.</param>
         /// <returns><c>true</c> if the project is ready, <c>false</c> if it is not.</returns>
         Task<bool> IsReadyAsync(IProject project, CancellationToken token);
+
+        /// <summary>
+        /// Gets a value indicating whether this check can be bypassed using the ReadinessAcknowledgement option.
```

Comment:
What about updating IUpgradeReadyCheck .IsReadyAsync to return an enum of `{ Ready, NotReady, Unsupported }` or something. It would allow for more states later on as well.